tests/test_dataset.py

```diff
@@ -48,17 +48,52 @@ def test_shape(self, expected_shape):
                 'extra': os.path.join(tempdir, 'test_extra2.nii.gz')
             }
         ]
-        dataset = Dataset(data=test_data, transform=LoadNiftid(keys=['image', 'label', 'extra']))
+        test_transform = Compose([LoadNiftid(keys=['image', 'label', 'extra']), TestIdentity(minimum_time_delay=1e-7)])
+
+        dataset = Dataset(data=test_data, transform=test_transform)
         data1 = dataset[0]
         data2 = dataset[1]
+
+        dataset = CacheDataset(data=test_data, transform=test_transform, num_workers=2)
```

Comment:
We already have 2 separate unit tests: `test_cachedataset.py` and `test_cachedataset_parallel.py`, so could you please keep this test for `Dataset` only and create another new unit test file for your `PersistentDataset`?
Thanks.